src/main/drivers/vtx_rtc6705.c

```diff
@@ -0,0 +1,247 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * Author: Giles Burgess (giles@multiflite.co.uk)
+ *
+ * This source code is provided as is and can be used/modified so long
+ * as this header is maintained with the file at all times.
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "platform.h"
+
+#include "common/maths.h"
+
+#include "drivers/vtx_rtc6705.h"
+#include "drivers/bus_spi.h"
+#include "drivers/system.h"
+#include "drivers/gpio.h"
+
+#define RTC6705_SET_HEAD 0x3210 //fosc=8mhz r=400
+#define RTC6705_SET_A1 0x8F3031 //5865
+#define RTC6705_SET_A2 0x8EB1B1 //5845
+#define RTC6705_SET_A3 0x8E3331 //5825
+#define RTC6705_SET_A4 0x8DB4B1 //5805
+#define RTC6705_SET_A5 0x8D3631 //5785
+#define RTC6705_SET_A6 0x8CB7B1 //5765
+#define RTC6705_SET_A7 0x8C4131 //5745
+#define RTC6705_SET_A8 0x8BC2B1 //5725
+#define RTC6705_SET_B1 0x8BF3B1 //5733
+#define RTC6705_SET_B2 0x8C6711 //5752
+#define RTC6705_SET_B3 0x8CE271 //5771
+#define RTC6705_SET_B4 0x8D55D1 //5790
+#define RTC6705_SET_B5 0x8DD131 //5809
+#define RTC6705_SET_B6 0x8E4491 //5828
+#define RTC6705_SET_B7 0x8EB7F1 //5847
+#define RTC6705_SET_B8 0x8F3351 //5866
+#define RTC6705_SET_E1 0x8B4431 //5705
+#define RTC6705_SET_E2 0x8AC5B1 //5685
+#define RTC6705_SET_E3 0x8A4731 //5665
+#define RTC6705_SET_E4 0x89D0B1 //5645
+#define RTC6705_SET_E5 0x8FA6B1 //5885
+#define RTC6705_SET_E6 0x902531 //5905
+#define RTC6705_SET_E7 0x90A3B1 //5925
+#define RTC6705_SET_E8 0x912231 //5945
+#define RTC6705_SET_F1 0x8C2191 //5740
+#define RTC6705_SET_F2 0x8CA011 //5760
+#define RTC6705_SET_F3 0x8D1691 //5780
+#define RTC6705_SET_F4 0x8D9511 //5800
+#define RTC6705_SET_F5 0x8E1391 //5820
+#define RTC6705_SET_F6 0x8E9211 //5840
+#define RTC6705_SET_F7 0x8F1091 //5860
+#define RTC6705_SET_F8 0x8F8711 //5880
+#define RTC6705_SET_R1 0x8A2151 //5658
+#define RTC6705_SET_R2 0x8B04F1 //5695
+#define RTC6705_SET_R3 0x8BF091 //5732
+#define RTC6705_SET_R4 0x8CD431 //5769
+#define RTC6705_SET_R5 0x8DB7D1 //5806
+#define RTC6705_SET_R6 0x8EA371 //5843
+#define RTC6705_SET_R7 0x8F8711 //5880
+#define RTC6705_SET_R8 0x9072B1 //5917
+
+#define RTC6705_SET_R  400     //Reference clock
+#define RTC6705_SET_FDIV 1024  //128*(fosc/1000000)
+#define RTC6705_SET_NDIV 16    //Remainder divider to get 'A' part of equation
+#define RTC6705_SET_WRITE 0x11 //10001b to write to register
+#define RTC6705_SET_DIVMULT 1000000 //Division value (to fit into a uint32_t) (Hz to MHz)
+
+#define DISABLE_RTC6705 GPIO_SetBits(RTC6705_CS_GPIO,   RTC6705_CS_PIN)
+#ifdef USE_RTC6705_CLK_HACK
+// HACK for missing pull up on CLK line - drive the CLK high *before* enabling the CS pin.
+#define ENABLE_RTC6705  {GPIO_SetBits(RTC6705_CLK_GPIO, RTC6705_CLK_PIN); delayMicroseconds(5); GPIO_ResetBits(RTC6705_CS_GPIO, RTC6705_CS_PIN); }
```

Comment:
Marked as hack, it is probably OK ... It may be better to convert it into function - it's quite unexpected when `ENABLE_` macro does something that long / nontrivial ...